library/std/src/rt.rs

```diff
@@ -45,9 +59,12 @@ fn lang_start<T: crate::process::Termination + 'static>(
     argc: isize,
     argv: *const *const u8,
 ) -> isize {
-    lang_start_internal(
-        &move || crate::sys_common::backtrace::__rust_begin_short_backtrace(main).report(),
-        argc,
-        argv,
-    )
+    unsafe {
+        // SAFETY: Err variant can't exist by definition.
+        Result::<_, !>::unwrap_unchecked(lang_start_internal(
```

Comment:
There is a `Result::into_ok` for this: https://doc.rust-lang.org/nightly/std/result/enum.Result.html#method.into_ok

(I suppose you could also use that directly instead of the `?`s above, to avoid changing the return type to Result.)